services/json-rpc/json_rpc_proxy.go

```diff
@@ -76,75 +68,30 @@ func (p *JsonRpcProxy) Start() error {
 func (p *JsonRpcProxy) metricHandler(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		t := time.Now()
-		agentConfig, foundAgent := p.findAgentFromRemoteAddr(req.RemoteAddr)
-		if foundAgent && p.rateLimiter.ExceedsLimit(agentConfig.ID) {
+		agentConfig, err := p.botAuthenticator.FindAgentFromRemoteAddr(req.RemoteAddr)
+		if err == nil && p.rateLimiter.ExceedsLimit(agentConfig.ID) {
```

Comment:
what if err exists?

Reply:
haven't modified the existing json-rpc proxy code. i guess it still executes the query?